src/test/java/com/connexta/ingest/service/impl/IngestServiceImplUnitTests.java

```diff
@@ -0,0 +1,360 @@
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class IngestServiceImplUnitTests {
+
+  @Mock private StoreClient mockStoreClient;
+  @Mock private MetacardStorageAdaptor mockMetacardStorageAdaptor;
+  @Mock private TransformClient mockTransformClient;
+
+  // TODO test invalid constructor arguments
+
+  @Test
+  public void testIngest() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    final URI location = new URI("http://localhost:9041/mis/product/1234");
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName)).thenReturn(location);
+
+    // when
+    assertDoesNotThrow(
+        () ->
+            ingestService.ingest(
+                fileSize, mimeType, inputStream, fileName, metacardFileSize, metacardInputStream));
+
+    // then
+    final InOrder inOrder =
+        Mockito.inOrder(mockStoreClient, mockMetacardStorageAdaptor, mockTransformClient);
+    // TODO Verify the key was generated as expected instead of anyString() and any(URI.class) here.
+    // Maybe extract a key generator.
+    inOrder
+        .verify(mockMetacardStorageAdaptor)
+        .store(eq(metacardFileSize), eq(METACARD_MEDIA_TYPE), eq(metacardInputStream), anyString());
+    inOrder
+        .verify(mockTransformClient)
+        .requestTransform(eq(location), eq(mimeType), any(URI.class));
+  }
+
+  /* START ingest StoreClient tests */
+
+  @Test
+  public void testIngestWhenStoreClientThrowsStoreException() {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final StoreException storeException = mock(StoreException.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
+        .thenThrow(storeException);
+
+    // when
+    final StoreException thrown =
+        assertThrows(
+            StoreException.class,
+            () ->
+                ingestService.ingest(
+                    fileSize, mimeType, inputStream, fileName, 1L, mock(InputStream.class)));
+
+    // then
+    assertThat(thrown, is(storeException));
```

Comment:
:pencil2: Is it important that the `thrown` object has the same identity as `storeException`? Is it enough to know that `thrown` is an instance of `StoreException`? If so, use  `assertThat(thrown, instanceOf(StoreException.class));`

Reply:
Yes, it's important because we use the exception to log and to include information in the error response.